api/x509cert_test.go

```diff
@@ -73,7 +74,10 @@ func TestGetX509CertificateAvailableSigningKeys(t *testing.T) {
 
 func TestGetX509CACertificate(t *testing.T) {
 	t.Parallel()
+	defaultCtx := context.Background()
+	timeoutCtx, cancel := context.WithTimeout(defaultCtx, 10*time.Microsecond)
```

Comment:
would it be possible to pick a more reasonable(longer) timeout, and have the function being exercised sleep for duration longer than that timeout?